src/io.c

```diff
@@ -0,0 +1,46 @@
+#include <assert.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "msemu.h"
+#include "sizes.h"
+
+/* Can be called multiple times, will zero the buffer if *io_buf is not null */
+/* Standard Z80 IO access will use lower 8bit of ADDR to be IO address with
+ * upper 8bit of ADDR being the value to write to the IO port
+ */
+int io_init(uint8_t **io_buf)
+{
+	if (*io_buf == NULL) {
+		*io_buf = (uint8_t *)calloc(SZ_256, sizeof(uint8_t));
+		if (*io_buf == NULL) {
+			printf("Unable to allocate IO buffer\n");
+			exit(EXIT_FAILURE);
+		}
+	} else {
+		/* Buffer is already allocated, just zero it out */
+		memset(*io_buf, '\0', SZ_256);
```

Comment:
Curious why you use `'\0'` here instead of just `0`. Never seen that before.

Reply:
Habit, mostly. memset() is a string manipulation function; normal use is to clear a string buffer and fill it with string terminators, aka '\0'